h/search/index.py

```diff
@@ -66,6 +66,44 @@ def index(self, annotation_ids: list, windowsize: int = PG_WINDOW_SIZE):
                 errored.add(status["_id"])
         return errored
 
+    def delete(self, annotation_ids: list[str]) -> None:
```

Comment:
In theory it should be possible to integrate annotation deletes into the `BatchIndexer.index()` method above, instead  of adding a separate `delete()` method.

In practice that's difficult because of how `BatchIndexer.index()` works: it streams annotations from a DB query into an Elasticsearch bulk API request. The problem is that annotations that've been marked as deleted and already purged from the DB aren't in the DB anymore, so they can't be streamed from a DB query.

It was easier to add a separate delete method.